Data-Structures/graph/graph.js

```diff
@@ -31,22 +30,47 @@ class Graph {
   }
 
   getNodes() {
-    if (!this.adjacencyList) return this.adjacencyList;
+    if (!this.adjacencyList.length) return this.adjacencyList;
```

Comment:
If `getNodes()` usually returns a new array, it should _always_ return a new array. A caller might assume it can safely modify the result, which would mess with your graph.

In other words you can just drop this check; `array.map(...)` for empty `array` returns a new empty array.